internal/resources/schemas/router_schemas.go

```diff
@@ -0,0 +1,225 @@
+package schemas
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func RouterTier0ConfigSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:          schema.TypeSet,
+		Optional:      true,
+		Description:   "Tier0 Gateway configuration",
+		MaxItems:      1,
+		ExactlyOneOf:  []string{"tier1_config", "tier0_config"},
+		ConflictsWith: []string{"tier1_config"},
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"ha_mode": {
+					Type:     schema.TypeString,
+					Required: true,
+					ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{
+						"ACTIVE_ACTIVE",
+					}, false)),
+					Description: "Available values are 'ACTIVE_ACTIVE'",
+				},
+				"fail_over": {
+					Type:     schema.TypeString,
+					Required: true,
+					ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{
+						"NON_PREEMPTIVE", "PREEMPTIVE",
+					}, false)),
+					Description: "Available values are 'NON_PREEMPTIVE'",
+				},
+				"enable_bgp": {
+					Type:         schema.TypeBool,
+					Optional:     true,
+					InputDefault: "true",
+				},
+				"route_redistribution_tier0": {
+					Type:     schema.TypeSet,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"tier0_static": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+							"tier0_nat": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+							"tier0_ipsec_local_ip": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+							"tier0_dns_forwarder_ip": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+							"tier0_service_interface": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+							"tier0_external_interface": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+							"tier0_loopback_interface": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+							"tier0_segment": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+						},
+					},
+				},
+				"route_redistribution_tier1": {
```

Comment:
Why is route redidtribution of tier1 is made availble in tier 0 schema?

Reply:
Same is available in UI as well.